src/Core.UnitTests/Notifications/DisabledNotificationsStorageTests.cs

```diff
@@ -18,19 +18,237 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using FluentAssertions;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Newtonsoft.Json;
 using SonarLint.VisualStudio.Core.Notifications;
+using SonarLint.VisualStudio.Core.VsVersion;
+using SonarLint.VisualStudio.Integration;
 using SonarLint.VisualStudio.Integration.UnitTests;
 
 namespace SonarLint.VisualStudio.Core.UnitTests.Notifications
 {
     [TestClass]
     public class DisabledNotificationsStorageTests
     {
+
         [TestMethod]
         public void MefCtor_CheckIsExported()
         {
-            MefTestHelpers.CheckTypeCanBeImported<DisabledNotificationsStorage, IDisabledNotificationsStorage>(null, null);
+            MefTestHelpers.CheckTypeCanBeImported<DisabledNotificationsStorage, IDisabledNotificationsStorage>(null, new[]
+            {
+                MefTestHelpers.CreateExport<IVsVersionProvider>(Mock.Of<IVsVersionProvider>()),
+                MefTestHelpers.CreateExport<ILogger>(Mock.Of<ILogger>())
+            });
+        }
+
+        [DataRow("2", true)]
+        [DataRow("4", false)]
+        [TestMethod]
+        public void IsNotificationDisabled_ReturnsCorrect(string notificationId, bool expectedResult)
+        {
+            var disabledNotifications = CreateDisabledNotifications("1", "2", "3");
+            var file = CreateFileMock(disabledNotifications);
+
+            var testSubject = CreateTestSubject(file.Object);
+                        
+            var result = testSubject.IsNotificationDisabled(notificationId);
+
+            result.Should().Be(expectedResult);
+            file.Verify(f => f.ReadAllText(GetFilePath()), Times.Once);
+        }
+
+        [TestMethod]
+        public void IsNotificationDisabled_ReadError_LogsError()
+        {
+            var file = new Mock<IFile>();
+            file.Setup(f => f.ReadAllText(It.IsAny<string>())).Throws(new Exception("Test"));
+            var logger = new TestLogger();
+
+            var testSubject = CreateTestSubject(file.Object, logger);
+
+            var result = testSubject.IsNotificationDisabled("1");
+
+            result.Should().BeFalse();
+            logger.AssertOutputStrings(2);
+            logger.AssertOutputStringExists("Failed to read disabled notification config from disk. Error:Test");
```

Comment:
We don't normally check for the exact text of localizable  strings.
Instead, we use a longer error message in the test exception and check that that appears in the output.

Search the existing code for `"this is a test"` - there are lots of examples.